pyscriptjs/tests/unit/pyodide.test.ts

```diff
@@ -1,18 +1,22 @@
 import type { AppConfig } from '../../src/pyconfig';
 import { InterpreterClient } from '../../src/interpreter_client';
-import { RemoteInterpreter } from '../../src/remote_interpreter';
 import { CaptureStdio } from '../../src/stdio';
 import * as Synclink from 'synclink';
-import { describe, beforeAll, afterAll, it, expect } from '@jest/globals';
+import { describe, beforeAll, afterAll, it, expect, jest } from '@jest/globals';
+import { dirname } from 'node:path';
+import type { RemoteInterpreter } from '../../src/remote_interpreter';
 
 describe('RemoteInterpreter', () => {
     let interpreter: InterpreterClient;
     let stdio: CaptureStdio = new CaptureStdio();
+    let RemoteInterpreter;
     const { port1, port2 } = new MessageChannel();
     beforeAll(async () => {
         const SRC = '../pyscriptjs/node_modules/pyodide/pyodide.js';
         const config: AppConfig = { interpreters: [{ src: SRC }] };
+        ({ RemoteInterpreter } = await import('../../src/remote_interpreter'));
```

Comment:
why not import this at the top?

Reply:
IIRC you shouldn't have issues importing dynamically outside once and then await the always same result.

```js
const RemoteInterpreterModule = import('../../src/remote_interpreter'));

// ... then
({ RemoteInterpreter } = await RemoteInterpreterModule);
```

although I don't see any concrete benefit or advantage in doing so, rather more comments to explain what's going on ... practically these two would do the same as the module is resolved once anyway, but I'd be curious to understand why we can't pre-mock whatever it is that we're mocking or why we couldn't use *importmaps* here to point at a different file/mock when needed.

Reply:
@hoodmane I think this *might* help if I fully understood what's the issue (jest mock not working top level) https://jestjs.io/docs/configuration#modulenamemapper-objectstring-string--arraystring

I've also found this [medium post](https://medium.com/@gokulc/configuring-module-resolution-on-typescript-and-jest-b17a93ebf54d) but I am afraid I don't know the author although it looks like that's the issue we're after?

Reply:
>  jest doesn't understand import maps ...

Sad face. Let's handle this in a followup!